roofit/roofitcore/inc/RooAddPdf.h

```diff
@@ -144,6 +144,13 @@ class RooAddPdf : public RooAbsPdf {
 
   mutable Int_t _coefErrCount ; //! Coefficient error counter
 
+  bool redirectServersHook(const RooAbsCollection&, bool, bool, bool) override {
+    // If a server is redirected, the cached normalization set might not point
+    // to the right observables anymore. We need to reset it.
+    _copyOfLastNormSet.reset();
+    return false;
+  } ;
```

Comment:
I would advise to use those opportunities to add a docstring. You have just figured out what the function does, so write it down immediately. 🙂 You will never be able to do it again without having to re-read the code first.

Also:
```suggestion
  bool redirectServersHook(const RooAbsCollection&, bool, bool, bool) override {
    // If a server is redirected, the cached normalization set might not point
    // to the right observables anymore. We need to reset it.
    _copyOfLastNormSet.reset();
    return false;
  }
```